coldfront/core/project/management/commands/deactivate_ica_projects.py

```diff
@@ -0,0 +1,236 @@
+            expiry_date = allocation.end_date
+
+            if expiry_date < current_date.date():
+                self.reset_service_units(project, options['dry_run'])
+                self.deactivate_project(project, allocation, options['dry_run'])
+
+                if options['send_emails']:
+                    self.send_emails(project, expiry_date, options['dry_run'])
+
+    def deactivate_project(self, project, allocation, dry_run):
+        """
+        Sets project status to Inactive and corresponding compute allocation to
+        Expired. Sets allocation start date to the current date and removes
+        the end date.
+
+        If dry_run is True, write to stdout without changing object fields.
+        """
+        project_status = ProjectStatusChoice.objects.get(name='Inactive')
+        allocation_status = AllocationStatusChoice.objects.get(name='Expired')
+        current_date = utc_now_offset_aware()
+
+        if dry_run:
+            message = (
+                f'Would update Project {project.name} ({project.pk})\'s '
+                f'status to {project_status.name} and Allocation '
+                f'{allocation.pk}\'s status to {allocation_status.name}.')
+
+            self.stdout.write(self.style.WARNING(message))
+        else:
+            project.status = project_status
+            project.save()
+            allocation.status = allocation_status
+            allocation.start_date = current_date
+            allocation.end_date = None
+            allocation.save()
+
+            message = (
+                f'Updated Project {project.name} ({project.pk})\'s status to '
+                f'{project_status.name} and Allocation {allocation.pk}\'s '
+                f'status to {allocation_status.name}.')
+
+            self.logger.info(message)
+            self.stdout.write(self.style.SUCCESS(message))
+
+    def reset_service_units(self, project, dry_run):
+        """
+        Resets service units for a project and its users to 0.00. Creates
+        the relevant transaction objects to record the change. Updates the
+        relevant historical objects with the reason for the SU change.
+
+        If dry_run is True, write to stdout without changing object fields.
+        """
+        allocation_objects = get_accounting_allocation_objects(project)
+        current_allocation = Decimal(allocation_objects.allocation_attribute.value)
+        current_date = utc_now_offset_aware()
+        reason = 'Resetting SUs while deactivating expired ICA project.'
+        updated_su = Decimal('0.00')
+
+        if dry_run:
+            message = f'Would reset {project.name} and its users\'s SUs from ' \
+                      f'{current_allocation} to {updated_su}. The reason ' \
+                      f'would be: "Resetting SUs while deactivating expired ' \
+                      f'ICA project."'
+
+            self.stdout.write(self.style.WARNING(message))
+
+        else:
+            # Set the value for the Project.
```

Comment:
Like the test code, this code looks to be the same as in #355. We probably want to have one or more helper functions.